app/src/main/java/com/hifeful/mealmania/presentation/myMeals/MyMealsFragment.kt

```diff
@@ -95,8 +96,13 @@ class MyMealsFragment : ObservableSourceFragment<MyMealsUiEvent>(), Consumer<MyM
         _binding = null
     }
 
-    override fun accept(viewState: MyMealsViewState) {
-        Log.d("MyMealsFragment", viewState.favouriteMeals.toString())
-        binding.bind(viewState, recentMealsAdapter, favouriteMealsAdapter)
+    private fun observeViewModel() {
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.state.collect {
+                    binding.bind(it, recentMealsAdapter, favouriteMealsAdapter)
+                }
+            }
+        }
```

Comment:
Correct ✅
Just FYI: there is the alternative of [Flow.flowWithLifecycle](https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda)

Reply:
👌 No action required